[REF] mail: rename web push models, clean code

tde-banana-odoo opened the pull request:
RATIONALE

Cleanup of enterprise code moved at odoo/odoo@0903ef7bb2117623cf6cbbd2375f56ca91babe6a

SPECIFICATIONS

Improve naming of push models: use 'mail.push' namespace

  * 'mail.notification.web.push' -> 'mail.push' is sufficient and understandable
    while being shorter;
  * 'mail.partner.device' -> 'mail.push.device' to use the mail.push namespace;

Update 'user_device' field name to 'mail_push_device_id' (both for new model
name and _id indicating it is a m2o).

Cleanup tests: move into 'test_mail' as it is now part of standard mail
code. 'test_mail_full' is used for integrations with advanced mail modules
like sms or mass mailing. Split into sub-tests to clarify tested features,
some tests are cleaned (notably using more explicit sub-tests).

Provide fixes spotted when improving / rewriting tests.

Task-3669738

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/148274).

awa-odoo reviewed (commented):
Bloups :fish: 

Some comments as promised, mostly small things like naming / typos / oopsies.

Cheers!

tde-banana-odoo commented on addons/test_mail/tests/test_mail_push.py:
```diff
@@ -321,8 +322,7 @@ def test_push_notifications_device_unreachable(self, post):
 
-
-    def test_push_notification_regenerate_vpaid_keys(self):
+    def test_push_notification_regenerate_vapid_keys(self):
```
I removed a duplicate blank line

tde-banana-odoo commented on addons/mail/models/mail_push_device.py:
```diff
@@ -42,44 +43,52 @@ def get_web_push_vapid_public_key(self):
-        sw_vapid_public_key = kw.get('vapid_public_key')
-        valid_sub = self._verify_vapid_public_key(sw_vapid_public_key)
+    def register_devices(self, endpoint=None, keys=None, vapid_public_key='',
+                         expirationTime=False, previousEndpoint=None):
```
Called by JS using a strange construct, don't want to retro engineer the JS so kept the camel case ^^ .

tde-banana-odoo commented on addons/mail/models/mail_push_device.py:
```diff
@@ -42,44 +43,52 @@ def get_web_push_vapid_public_key(self):
+        }
+        if not mail_push_device:
+            self.sudo().create(update_values)
+        elif mail_push_device.partner_id is not self.env.user.partner_id:
```
Indeed.

tde-banana-odoo reviewed (commented):
Updated ! Note that I moved the cleaning itself in another branch (https://github.com/odoo/odoo/pull/148274), to try to make the renaming land on time for the freeze (cleanup can be finished later).

tde-banana-odoo commented:
@rfr-odoo @res-odoo FYI, cleaned a bit web push tests in mail, and renamed models to better match mail naming :)  While updating tests I discovered some small issues, hence some fixes (will probably backport them in 17).

Feel free to comment !

Zboing !

awa-odoo commented on addons/test_mail/tests/test_mail_push.py:
```diff
@@ -86,55 +62,75 @@ def _assert_notification_count_for_cron(self, number_of_notification):
+    def test_notify_by_push(self, push_to_end_point):
+        """ In case of notification, only inbox users are notified """
+        for recipient, has_notification in [(self.user_email, False), (self.user_inbox, True)]:
+            with self.subTest(recipient=self.user_email):
```
Ok but you force `recipient=self.user_email` instead of the one used in the current loop iteration though, which does not feel right, unless I misunderstood the concept of subTest.

tde-banana-odoo replied:
Ah yes.

tde-banana-odoo commented on addons/mail/models/mail_thread.py:
```diff
@@ -3663,7 +3663,7 @@ def _notify_by_web_push_prepare_payload(self, message, msg_vals=False):
         if author_name:
             title = "%s: %s" % (author_name, title)
-            icon = "/web/image/res.users/%d/avatar_128" % author_id[0]
+            icon = "/web/image/res.partner/%d/avatar_128" % author_id[0]
```
Yep, fixes will be backported :)

tde-banana-odoo commented:
@robodoo override=ci/security

Reason: just using markup in tests to check equality in post flows

tde-banana-odoo commented:
@robodoo r+ rebase-merge

To people: merged as freeze is upon us (in order to close the renaming), any discussion can continue on https://github.com/odoo/odoo/pull/149088

robodoo commented:
Merge method set to rebase and merge, using the PR as merge commit message.

robodoo commented:
@tde-banana-odoo linked pull request(s) odoo/enterprise#53725 not ready. Linked PRs are not staged until all of them are ready.

nd-dew commented:
previous modifications/renamings:
https://github.com/odoo/odoo/pull/148179
